src/libasr/pass/replace_symbolic.cpp

```diff
@@ -0,0 +1,178 @@
+                    ASR::stmt_t* stmt2 = ASRUtils::STMT(ASR::make_Assignment_t(al, xx.base.base.loc, target2, value2, nullptr));
+                    ASR::stmt_t* stmt3 = ASRUtils::STMT(ASR::make_Assignment_t(al, xx.base.base.loc, target2, value3, nullptr));
+                    //ASR::stmt_t* stmt4 = ASRUtils::STMT(ASR::make_SubroutineCall_t(al, xx.base.base.loc, basic_new_stack_sym, nullptr, call_args.p, call_args.n, nullptr));
+
+                    // push stmt1 into the updated body vector
+                    pass_result.push_back(al, stmt1);
+                    pass_result.push_back(al, stmt2);
+                    pass_result.push_back(al, stmt3);
+                    //pass_result.push_back(al, stmt4);
+
+                    // updated x.m_body and x.n_bdoy with that of the updated vector
+                    // x.m_body = updated_body.p;
+                    // x.n_body = updated_body.size();
+                    transform_stmts(xx.m_body, xx.n_body);
+                    symbolic_replaces_with_CPtr_Function = false;
+                }
+            }
+        }
+        current_scope = current_scope_copy;
+    }
+
+    void visit_Variable(const ASR::Variable_t& x) {
+        SymbolTable* current_scope_copy = current_scope;
+        current_scope = x.m_parent_symtab;
+        if (x.m_type->type == ASR::ttypeType::SymbolicExpression) {
+            symbolic_replaces_with_CPtr_Module = true;
+            symbolic_replaces_with_CPtr_Function = true;
+            std::string var_name = x.m_name;
+            std::string placeholder = "_" + std::string(x.m_name);
+
+            // defining CPtr variable
+            ASR::ttype_t *type1 = ASRUtils::TYPE(ASR::make_CPtr_t(al, x.base.base.loc));
+            ASR::symbol_t* sym1 = ASR::down_cast<ASR::symbol_t>(
+                ASR::make_Variable_t(al, x.base.base.loc, current_scope,
+                                    s2c(al, var_name), nullptr, 0,
+                                    x.m_intent, nullptr,
+                                    nullptr, x.m_storage,
+                                    type1, nullptr, x.m_abi,
+                                    x.m_access, x.m_presence,
+                                    x.m_value_attr));
+
+            ASR::ttype_t *type2 = ASRUtils::TYPE(ASR::make_Integer_t(al, x.base.base.loc, 8));
+            ASR::symbol_t* sym2 = ASR::down_cast<ASR::symbol_t>(
+                ASR::make_Variable_t(al, x.base.base.loc, current_scope,
+                                    s2c(al, placeholder), nullptr, 0,
+                                    x.m_intent, nullptr,
+                                    nullptr, x.m_storage,
+                                    type2, nullptr, x.m_abi,
+                                    x.m_access, x.m_presence,
+                                    x.m_value_attr));
+
+            current_scope->erase_symbol(s2c(al, var_name));
+            current_scope->add_symbol(s2c(al, var_name), sym1);
```

Comment:
Instead of erasing and adding the symbol, I think we can modify the type. See if it works?